File: Dataset/Leetcode/train/100/274.cpp
 class Solution {
public:
    bool XXX(TreeNode* p, TreeNode* q) {
        if(p != nullptr && q != nullptr) return XXX(p->left, q->left) && (p->val == q->val) && XXX(p->right, q->right);
        else if(p == nullptr && q == nullptr) return true;
        else return false;
    }
};

File: Dataset/Leetcode/train/101/274.java
 class Solution {
    public boolean XXX(TreeNode root) {
        return comp(root, root);
    }
    
    public boolean comp(TreeNode node1, TreeNode node2) {
        if (node1 == null && node2 == null) {
            return true;
        }
        if (node1 == null || node2 == null || node1.val != node2.val) {
            return false;
        }
        return comp(node1.left, node2.right) && comp(node1.right, node2.left);
    }
}

